controllers/csiaddonsnode_controller.go

```diff
@@ -39,17 +51,87 @@ type CSIAddonsNodeReconciler struct {
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CSIAddonsNode object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
 func (r *CSIAddonsNodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx)
+
+	// Fetch CSIAddonsNode instance
+	csiAddonsNode := &csiaddonsv1alpha1.CSIAddonsNode{}
+	err := r.Client.Get(ctx, req.NamespacedName, csiAddonsNode)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			logger.Info("CSIAddonsNode resource not found")
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	err = validateCSIAddonsNodeSpec(csiAddonsNode)
+	if err != nil {
+		logger.Error(err, "Failed to validate CSIAddonsNode parameters")
+
+		csiAddonsNode.Status.State = csiaddonsv1alpha1.CSIAddonsNodeStateFailed
+		csiAddonsNode.Status.Message = fmt.Sprintf("Failed to validate CSIAddonsNode parameters: %v", err)
+		statusErr := r.Client.Status().Update(ctx, csiAddonsNode)
+		if statusErr != nil {
+			logger.Error(statusErr, "Failed to update status")
+
+			return ctrl.Result{}, statusErr
+		}
+
+		// invalid parameter, do not requeue
+		return ctrl.Result{}, nil
```

Comment:
In case parameters are invalid/empty... do we not want to return an error?

suggested change `return ctrl.Result{}, err`
Thoughts?

Reply:
> In case parameters are invalid/empty... do we not want to return an error?
> 
> suggested change `return ctrl.Result{}, err` Thoughts?

it is intentional, we're logging the error and dequeue it. 
No use in reconciling invalid parameters again and again. 

If there is a change in parameter from outside, it'll be requeued for reconcile

Reply:
Cool :+1: